components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java

```diff
@@ -92,6 +94,297 @@
+                    Map<String, Object> defTypeExtension = defaultTypeFlow.getVendorExtensions();
+                    if (notdefFlowType.getVendorExtensions() != null && (notdefScopeBindings =
+                            (Map<String, String>) notdefFlowType.getVendorExtensions().get(APIConstants.SWAGGER_X_SCOPES_BINDINGS))
+                            != null) {
+                        if(defScopeBindings == null) {
+                            defScopeBindings = new HashMap<>();
+                        }
+                        for (Map.Entry<String, String> roleInUse : notdefScopeBindings.entrySet()) {
+                            String notdeftypescope = roleInUse.getKey();
+                            String notdeftypeRole = roleInUse.getValue();
+                            defScopeBindings.put(notdeftypescope, notdeftypeRole);
+                        }
+                    }
+                    defTypeExtension.put(APIConstants.SWAGGER_X_SCOPES_BINDINGS, defScopeBindings);
+                    defaultTypeFlow.setVendorExtensions(defTypeExtension);
+                    securityDefinitions.put(SWAGGER_SECURITY_SCHEMA_KEY, defaultTypeFlow);
+
+                }
+            }
+        }
+        swagger.setSecurityDefinitions(securityDefinitions);
+        if (securityDefinitions != null
+                && (oAuth2Definition = (OAuth2Definition) securityDefinitions.get(SWAGGER_SECURITY_SCHEMA_KEY)) != null
+                && oAuth2Definition.getScopes() != null) {
+            for (Map.Entry<String, String> entry : oAuth2Definition.getScopes().entrySet()) {
+                Scope scope = new Scope();
+                scope.setKey(entry.getKey());
+                scope.setName(entry.getKey());
+                scope.setDescription(entry.getValue());
+                Map<String, String> scopeBindings;
+                if (oAuth2Definition.getVendorExtensions() != null && (scopeBindings =
+                        (Map<String, String>) oAuth2Definition.getVendorExtensions()
+                                .get(APIConstants.SWAGGER_X_SCOPES_BINDINGS)) != null) {
+                    if (scopeBindings.get(scope.getKey()) != null) {
+                        scope.setRoles(scopeBindings.get(scope.getKey()));
+                    }
+                }
+                scopeSet.add(scope);
+            }
+            setOtherSchemes(otherSetofSchemes);
+            return OASParserUtil.sortScopes(scopeSet);
+        } else {
+            setOtherSchemes(otherSetofSchemes);
+            return OASParserUtil.sortScopes(getScopesFromExtensions(swagger));
+        }
+    }
+
+    /**
+     * This method returns URI templates according to the given swagger file(Swagger version 2)
+     *
+     * @param resourceConfigsJSON swaggerJSON
+     * @return URI Templates
+     * @throws APIManagementException
+     */
+    @Override
+    public Set<URITemplate> injectOtherRescouceScopesToDefault(String resourceConfigsJSON) throws APIManagementException {
```

Comment:
Fix the small typo in the method. Shall we make it injectOtherResourceScopesToDefaultScheme